### DISCUSSION TITLE
Add random noise to edge attribute

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#2578] [Q&A]
How can I add random noise to the edge attribute before passing it into the GCN layer?

**Comment by rusty1s** [comment]
I guess you can simply add random Gaussian noise to them, e.g.:
```python
edge_attr = edge_attr + 0.01 * torch.randn_like(edge_attr)
```

**Reply by shrimonmuke0202** [reply]
I want to add trainable random noise to the edge_attribute?

**Reply by shrimonmuke0202** [reply]
How to add random gaussian noise to the adjacency matrix of a graph?

**Reply by rusty1s** [reply]
We already answered this here: https://github.com/rusty1s/pytorch_geometric/discussions/2493#discussioncomment-673292
